### ISSUE TITLE
Can’t build on AArch32 CPU architecture (Raspberry Pi)

### ISSUE THREAD

**Issue by alandao** [jonathandturner/rawkey#5]
Hi there,

I’m trying to get NuShell working on ARM. This package is listed as a dependency and it doesn’t build on my Raspberry Pi 4 which is running Raspbian 32-bit(armv71/AArch32). This is the error I’m getting.

```
Compiling rawkey v0.1.2                                                                                                                          
error[E0308]: mismatched types                                                                                                                      
  --> /home/pi/.cargo/registry/src/github.com-1ecc6299db9ec823/rawkey-0.1.2/src/linux.rs:44:46
   |                                                                                                                                                
44 |             xlib::XQueryKeymap(self.display, keymap);                                                                                          
   |                                              ^^^^^^ expected `u8`, found `i8`
   |                                                                                                                                                
   = note: expected raw pointer `*mut u8`                                                                                                           
              found raw pointer `*mut i8`                                                                                                           
                                                                                                                                                    
error: aborting due to previous error                                                                                                               
                                                                                                                                                    
For more information about this error, try `rustc --explain E0308`.                                                                                 
error: could not compile `rawkey`.                                                                                                                  
                                                                                                                                                    
To learn more, run the command again with --verbose.                                                                                                
warning: build failed, waiting for other jobs to finish...                                                                                          
error: build failed                                    
```

I found a resolution for a similar issue in https://github.com/remacs/remacs/issues/1393. Could we try that fix here and merge it in?

**Comment by hampuslidin** [comment]
Hi @jonathandturner, is this something which we could move forward with? PR #3 has addressed this issue by letting Rust infer the type. Alternatively, I have made a fork which adds libc as a dependency for Linux (which was a sub-dependency anyway, so it doesn't add any build time), and uses *mut libc::c_char as the keymap pointer type. Thanks in advance!

**Comment by hampuslidin** [comment]
This was fixed by #3.
